spec/forms/otp_delivery_selection_form_spec.rb

```diff
@@ -42,5 +42,32 @@
         expect(result).to eq result_hash
       end
     end
+
+    context 'when otp_method is the same as the user otp_delivery_preference' do
+      it 'does not update the user' do
+        user = build_stubbed(:user, otp_delivery_preference: 'sms')
+        form = OtpDeliverySelectionForm.new(user)
+
+        expect(UpdateUser).to_not receive(:new)
```

Comment:
I thought that we did that only if there are more than 1 expectation, that way they are grouped together. If there is only one expectation, I prefer the more succinct form, but I'm happy to change it if we think we should always use the allow+expect pattern.